applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -41,9 +41,148 @@ array_1d<double, NumNodes> GetPotentialOnNormalElement(const Element& rElement)
+    upper_potentials = GetPotentialOnUpperWakeElement<Dim, NumNodes>(rElement, rDistances);
+
+    array_1d<double, NumNodes> lower_potentials;
+    lower_potentials = GetPotentialOnLowerWakeElement<Dim, NumNodes>(rElement, rDistances);
+
+    BoundedVector<double, 2 * NumNodes> split_element_values;
+    for (unsigned int i = 0; i < NumNodes; i++) {
+        split_element_values[i] = upper_potentials[i];
+        split_element_values[NumNodes + i] = lower_potentials[i];
+    }
+
+    return split_element_values;
+}
+
+template <int Dim, int NumNodes>
+BoundedVector<double, NumNodes> GetPotentialOnUpperWakeElement(
+    const Element& rElement, const array_1d<double, NumNodes>& rDistances)
+{
+    array_1d<double, NumNodes> upper_potentials;
+    const auto r_geometry = rElement.GetGeometry();
+
+    for (unsigned int i = 0; i < NumNodes; i++){
+        if (rDistances[i] > 0){
+            upper_potentials[i] = r_geometry[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+        }
+        else{
+            upper_potentials[i] = r_geometry[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        }
+    }
+
+    return upper_potentials;
+}
+
+template <int Dim, int NumNodes>
+BoundedVector<double, NumNodes> GetPotentialOnLowerWakeElement(
+    const Element& rElement, const array_1d<double, NumNodes>& rDistances)
+{
+    array_1d<double, NumNodes> lower_potentials;
+    const auto r_geometry = rElement.GetGeometry();
+
+    for (unsigned int i = 0; i < NumNodes; i++){
+        if (rDistances[i] < 0){
+            lower_potentials[i] = r_geometry[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+        }
+        else{
+            lower_potentials[i] = r_geometry[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        }
+    }
+
+    return lower_potentials;
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocity(const Element& rElement)
+{
+    const int wake = rElement.GetValue(WAKE);
+
+    if (wake == 0)
+        return PotentialFlowUtilities::ComputeVelocityNormalElement<Dim,NumNodes>(rElement);
+    else
+        return PotentialFlowUtilities::ComputeVelocityUpperWakeElement<Dim,NumNodes>(rElement);
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocityNormalElement(const Element& rElement)
+{
+    ElementalData<NumNodes, Dim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(rElement.GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    data.potentials = GetPotentialOnNormalElement<Dim,NumNodes>(rElement);
+
+    return prod(trans(data.DN_DX), data.potentials);
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocityUpperWakeElement(const Element& rElement)
+{
+    ElementalData<NumNodes, Dim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(rElement.GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    array_1d<double, NumNodes> distances = rElement.GetValue(ELEMENTAL_DISTANCES);
```

Comment:
Here I would use a function like: GetWakeDistances()

Reply:
Even if it is a one line function?

Reply:
My main point would be to have the way we retrieve the distance of the wake  in only one place. If in future we retrieve the distance from another variable name or from a nodal variable we would only need to change it once. 

Of course it’s minor if you think it’s not worth it’s ok!

Reply:
That is a good point, I will change it. Thanks!